src/main/resources/templates/SongIndex.html

```diff
@@ -0,0 +1,23 @@
+<!DOCTYPE HTML>
+<html xmlns:th="http://www.thymeleaf.org">
+<head>
+    <meta charset="UTF-8">
+    <title>Song</title>
+</head>
+ <body>
+ <ul>
+        <h2 th: text="${song.title}"></h2>

```

Comment:
This doesn't connect with the variables you created in your controller, nor does the name of this file line up with the string you return there. Even if any of this code were running, it wouldn't be working correctly.